File: src/visual-graph/builder/should-subgraph.ts
import type { SerializedScope } from "../../ir/serialized/serialized-scope.js";
import { isClassSubgraph } from "./is-class-subgraph.js";
import { isControlSubgraph } from "./is-control-subgraph.js";
import { isFunctionSubgraph } from "./is-function-subgraph.js";

export function shouldSubgraph(scope: SerializedScope): boolean {
  return (
    isFunctionSubgraph(scope) ||
    isClassSubgraph(scope) ||
    isControlSubgraph(scope)
  );
}

File: crates/unsnarl-visual-graph/src/builder/should_subgraph.rs
//! Mirrors `ts/src/visual-graph/builder/should-subgraph.ts`.

use unsnarl_ir::serialized::SerializedScope;

use super::is_class_subgraph::is_class_subgraph;
use super::is_control_subgraph::is_control_subgraph;
use super::is_function_subgraph::is_function_subgraph;

pub fn should_subgraph(scope: &SerializedScope) -> bool {
    is_function_subgraph(scope) || is_class_subgraph(scope) || is_control_subgraph(scope)
}

#[cfg(test)]
#[path = "should_subgraph_test.rs"]
mod should_subgraph_test;
